src/EigenAirdrop.sol

```diff
@@ -0,0 +1,266 @@
+    IStrategyManager public strategyManager;
+
+    /// @notice Emitted when a user claims tokens from the airdrop.
+    /// @param user The address of the user claiming tokens.
+    /// @param amount The amount of tokens claimed.
+    event Claimed(address user, uint256 amount);
+
+    /// @notice Emitted when a user claims and restakes tokens.
+    /// @param user The address of the user.
+    /// @param amount The amount of tokens restaked.
+    /// @param shares The amount of shares received from restaking.
+    event ClaimedAndRestaked(address user, uint256 amount, uint256 shares);
+
+    /// @notice Error thrown when there is no airdrop available.
+    error NoAirdrop();
+    error InvalidAirdrop();
+    error DeadlinePassed();
+    error InvalidInit();
+
+    /**
+     * @dev Modifier to check if the user can claim the specified amount.
+     * Reverts if the amount is zero or exceeds the claimable balance.
+     * @param _amountToClaim The amount the user is trying to claim.
+     */
+    modifier whenAvailable(uint256 _amountToClaim) {
+        if (block.timestamp > deadline) {
+            revert DeadlinePassed();
+        }
+        if (_amountToClaim == 0 || _amountToClaim > amounts[msg.sender]) {
+            revert NoAirdrop();
+        }
+        _;
+        amounts[msg.sender] -= _amountToClaim;
+        totalAmount -= _amountToClaim;
+    }
+
+    /**
+     * @dev Disables initializers to prevent contract from being reinitialized.
+     */
+    constructor() {
+        _disableInitializers();
+    }
+
+    /**
+     * @notice Initializes the airdrop contract with the provided parameters.
+     * @param _owner The address of the owner.
+     * @param _safe The address of the safe holding the tokens.
+     * @param _token The address of the token being airdropped.
+     * @param _strategy The address of the strategy for restaking.
+     * @param _strategyManager The address of the strategy manager.
+     * @param _deadline The timestamp on which the claims are no longer valid.
+     * @param _userAmounts An array of user amounts for the airdrop.
+     */
+    function initialize(
+        address _owner,
+        address _safe,
+        address _token,
+        address _strategy,
+        address _strategyManager,
+        uint256 _deadline,
+        UserAmount[] memory _userAmounts
+    )
+        public
+        initializer
+    {
+        __Pausable_init();
+        __Ownable_init(_owner);
+        __ReentrancyGuard_init();
+
+        if (
+            _safe == address(0) || _token == address(0) || _strategy == address(0)
+                || _strategyManager == address(0) || _deadline == 0
+        ) {
+            revert InvalidInit();
+        }
+
+        safe = _safe;
+        deadline = _deadline;
+        token = IERC20(_token);
+        strategy = IStrategy(_strategy);
+        strategyManager = IStrategyManager(_strategyManager);
+
+        _updateUserAmounts(_userAmounts);
```

Comment:
from gas perspective, how much gas does it use to load the existing entries in the sheet?

Can we easily fit this in a block?

Reply:
I'll take a look and respond with numbers. But simulation seems to work with the current sheet.